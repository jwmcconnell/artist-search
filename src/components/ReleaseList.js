import React from 'react';
import PropTypes from 'prop-types';
import ReleaseItem from './ReleaseItem';

const styles = {
  list: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'center',
    listStyle: 'none',
    margin: 0,
    padding: 0
  }
};

const ReleaseList = ({ releasesData }) => {
  console.log(releasesData);
  const releases = releasesData.map(release => (
    <ReleaseItem key={release.id} release={release} />
  ));
  return (
    <ul style={styles.list}>{releases}</ul>
  );
};

ReleaseList.propTypes = {
  releasesData: PropTypes.array.isRequired
};

export default ReleaseList;
