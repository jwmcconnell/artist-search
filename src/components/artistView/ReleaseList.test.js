import React from 'react';
import { shallow } from 'enzyme';
import ReleaseList from './ReleaseList';

describe('ReleaseList component', () => {
  it('renders ReleaseList', () => {
    const wrapper = shallow(<ReleaseList releasesData={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
