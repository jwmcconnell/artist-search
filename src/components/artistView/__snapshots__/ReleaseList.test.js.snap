// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReleaseList component renders ReleaseList 1`] = `
<ul
  style={
    Object {
      "display": "flex",
      "flexWrap": "wrap",
      "justifyContent": "center",
      "listStyle": "none",
      "margin": 0,
      "padding": 0,
    }
  }
/>
`;
